### PR TITLE
Spring hateoas 1.0

### PR DESCRIPTION
**Pull request by ericenns**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Upgraded spring-hateoas from 0.19.0 to 1.0.3. This is required for migration of the application over to Sprint-Boot 2.2.X.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
~~* [ ] Tests added (or description of how to test) for any new features.~~
~~* [ ] User documentation updated for UI or technical changes.~~

### CODE REVIEW

**Review by ksierks — APPROVED** [review]
Awesome! Curious if you used the migrate-to-1.0.sh script? I tested random API endpoints, including the content negotiator. Looks good to me.

**Comment by ericenns** [comment]
@ksierks I did end up using the `migrate-to-1.0.sh` script, just not on the entire repo only on the files that actually use `spring-hateoas`.

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Just a few questions and changes. Otherwise looking good

**SampleSequenceFilesControllerTest.java**

Line 218: assertNotNull(qc.getLink(RESTSampleSequenceFilesController.REL_QC_SEQFILE));

**RESTSampleAssemblyControllerTest**

Line 76: assertNotNull("has self rel", genomeAssembly.getLink("self"));
Line 87: assertNotNull("has self rel", genomeAssembly.getLink("self"));

**RootControllerTest**

Line 50: assertNotNull(r.getLink("self"));
Line 51: assertNotNull(r.getLink("version"));

Is the  `.map(i -> i)`.orElse(null) not required for some of these and IanaLinkRelations.SELF for some of them?

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Changes
 * [UI]: Refactored share/move samples between projects to use a separate page.
 * [UI]: Added capability to share/move samples between projects from the line list page.
 * [UI]: Fixed issue with dynamic pipeline parameters with only one value not rendering.
+* [DEVELOPER]: Upgraded Spring Hateoas to 1.0.3.RELEASE
```
[the remark this answers is not in the document]
Comment (ericenns) [reply CHANGELOG.md]: Resolved in 4dc1f5effd6bbc9925f8440d57bcac7c4b4a6c5b

[the remark this answers is not in the document]
Comment (deepsidhu85) [reply CHANGELOG.md]: Thanks for taking care of that!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**